LAD-AMA-Common/metrics_ext_utils/metrics_ext_handler.py

```diff
@@ -94,6 +94,8 @@ def stop_metrics_service(is_lad):
 
         if os.path.isfile(metrics_service_path):
             code = os.system("sudo systemctl stop {0}".format(metrics_service_name))
+            if code != 0:
+                code = os.system("systemctl stop {0}".format(metrics_service_name))
```

Comment:
do we even need sudo? I do not see it used in `agent.py`

Reply:
this code is shared with LAD, I need to check if LAD needs sudo

Reply:
Nikhil added the sudo, should we just remove it and see if it breaks anything? :) I checked LAD code and that should be running
 as root as well so sudo shouldn't be needed

Reply:
They can have the opposite when the user is not the root, in this case it should run with sudo.

Reply:
user is always root, Guest agent invokes our extension code with root

Reply:
let me remove sudo and I feel we should be fine

Reply:
waagent may be, but we are using shim to apply manual workarounds. Is this code also called from shim?

Reply:
manual workaround can do sudo shim.sh instead...that is for internal use only

Reply:
ok, but we will need to check with CSS tsg and update the steps . I am almost certain we have shim in troubleshooting.